scripts/tools/memory/gh_report.py

```diff
@@ -0,0 +1,548 @@
+        origin = {'file': filename}
+        path = Path(filename)
+        if path.suffix == '.json':
+            logging.info('Reading JSON %s', path)
+            with open(path) as f:
+                self.add_sizes_from_json(f.read(), origin)
+        elif path.suffix == '.zip':
+            logging.info('Reading ZIP %s', path)
+            self.add_sizes_from_zipfile(path, origin)
+        else:
+            logging.warning('Unknown file type "%s" ignored', filename)
+
+    def add_sizes_from_github(self):
+        """Read size report artifacts from github."""
+        if not self.gh:
+            return
+
+        # Size artifacts have names of the form
+        #   Size,{group},{pr},{commit_hash},{parent_hash}
+        # Record them keyed by group and commit_hash to match them up
+        # after we have the entire list.
+        size_artifacts: Dict[str, Dict[str, fastcore.basics.AttrDict]] = {}
+        for i in ghapi.all.paged(self.gh.actions.list_artifacts_for_repo):
+            if not i.artifacts:
+                break
+            for a in i.artifacts:
+                if a.name.startswith('Size,'):
+                    _, group, pr, commit, parent, *_ = (a.name + ',').split(
+                        ',', 5)
+                    a.parent = parent
+                    a.pr = pr
+                    a.created_at = dateutil.parser.isoparse(a.created_at)
+                    if group not in size_artifacts:
+                        size_artifacts[group] = {}
+                    size_artifacts[group][commit] = a
+
+        # Determine required size artifacts.
+        required_artifact_ids: set[int] = set()
+        for group, group_reports in size_artifacts.items():
+            logging.info('Group %s', group)
+            for report in group_reports.values():
+                if self.config['report.pr' if report.pr else 'report.push']:
+                    if report.parent not in group_reports:
+                        logging.info('  No match for %s', report.name)
+                        continue
+                    # We have size information for both this report and its
+                    # parent, so ensure that both artifacts are downloaded.
+                    parent = group_reports[report.parent]
+                    required_artifact_ids.add(report.id)
+                    required_artifact_ids.add(parent.id)
+                    logging.info('  Match %s', report.parent)
+                    logging.info('    %s %s', report.id, report.name)
+                    logging.info('    %s %s', parent.id, parent.name)
+
+        # Download and add required artifacts.
+        for i in required_artifact_ids:
+            logging.debug('Download artifact %d', i)
+            try:
+                blob = self.gh.actions.download_artifact(i, 'zip')
+            except Exception as e:
+                logging.error('Failed to download artifact %d: %s', i, e)
+            self.add_sizes_from_zipfile(io.BytesIO(blob), {'artifact': i})
```

Comment:
But some of the artifacts we uploaded were a single JSON file, not zipfiles, no?  See `examples-infineon.yaml`.

Reply:
Github unconditionally wraps workflows artifacts in a zip; there's no other way to get them.